subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/assignment/internal/KotlinDslAssignment.kt

```diff
@@ -16,21 +16,10 @@
 
 package org.gradle.kotlin.dsl.assignment.internal
 
-import org.gradle.util.internal.IncubationLogger
-
 
+/**
+ * TODO: Remove once kotlin-dsl plugin is updated
+ */
 object KotlinDslAssignment {
```

Comment:
🤔 In order to keep some backwards compatibility for folks mixing up `kotlin-dsl` plugin versions we could:
- mark this type as `@Deprecated(HIDDEN)`
- let `isAssignmentOverloadEnabled()` emit an actionable deprecation warning
- remove this in 9.0

WDYT?

Reply:
That sounds good idea, I will do that

Reply:
It turns out that if I use `@Deprecated(HIDDEN)` compilation still fails if you have warnings as errors. So I just added a deprecation warning.

Reply:
A lot of tests fail when I add a deprecation warning to KotlinDslAssignment. It seems easier if we update `kotlin-dsl` first and then add a deprecation warning. I opened an issue: https://github.com/gradle/gradle/issues/25742